bug: agent run shows success but exits with status 1, checkpoint fails silently

## Summary

Agent runs that complete Claude Code execution successfully are reporting `exit status 1` failure. The CLI output shows `[result] ✓ completed successfully` but immediately followed by `[vm0_error] ✗ Run failed`. The root cause appears to be checkpoint creation failing, but the error logs don't provide enough detail to diagnose why.

## Reproduction Steps

1. Run an agent with an artifact:
   ```bash
   vm0 run standard-agent "answer my question" --artifact-name user-files
   ```

2. Observe the CLI output shows success then failure:
   ```
   [result] ✓ completed successfully
     Duration: 3.5s
     Cost: $0.0518
   [vm0_error] ✗ Run failed
     Error: exit status 1
   ```

## Expected Behavior

- If Claude Code completes successfully, the run should succeed
- If checkpoint creation fails, specific error details should be logged

## Actual Behavior

- Claude Code completes successfully (`[result] ✓ completed successfully`)
- Run immediately fails with `exit status 1`
- Web logs show only generic `CommandExitError: exit status 1` with no details
- No indication of what specifically failed (checkpoint, VAS snapshot, API call, etc.)

## Error Logs

**CLI output:**
```
[vm0_start] Run starting
[init] Starting Claude Code agent
[text] I'd be happy to help! What question would you like me to answer?
[result] ✓ completed successfully
  Duration: 3.5s
[vm0_error] ✗ Run failed
  Error: exit status 1
```

**Web server logs:**
```
2025-11-28 08:34:18.426 [error] [E2B] Run e4f23b51-e6d0-4501-911a-b14ca400a4eb failed: Error [CommandExitError]: exit status 1
    at ee.wait (.next/server/app/api/agent/runs/route.js:1:227708)
```

## Analysis

Looking at `run-agent.ts` (the sandbox script), after Claude Code succeeds:

```bash
if [ $CLAUDE_EXIT_CODE -eq 0 ]; then
  log_info "Claude Code completed successfully"
  
  # Create checkpoint - this is mandatory for successful runs
  if ! create_checkpoint; then
    log_error "Checkpoint creation failed, marking run as failed"
    exit 1
  fi
fi
```

The checkpoint creation (`create_checkpoint` function) could fail for multiple reasons:
- Missing session ID file
- Missing session history path file  
- Session history file not found
- Empty session history
- Artifact not configured (required)
- VAS snapshot creation failed
- Checkpoint API call failed

The current logging in sandbox scripts outputs to stderr with `[INFO]/[ERROR] [sandbox:run-agent]` format, but these logs are captured in `result.stderr` and only logged at debug level in `e2b-service.ts:500`:

```typescript
log.debug(`stderr (${result.stderr.length} chars):`, result.stderr);
```

## Root Causes

1. **Insufficient error logging**: Sandbox stderr output (which contains the actual error) is only logged at debug level
2. **Missing error propagation**: When checkpoint fails, we only know `exit 1` happened, not why
3. **Silent failures**: The checkpoint function has many failure points but the specific error isn't surfaced to the user

## Suggested Fix

1. **Log stderr at error level when exit code != 0** in `e2b-service.ts`:
   ```typescript
   if (result.exitCode !== 0) {
     log.error(`Run ${runId} failed with exit code ${result.exitCode}`);
     log.error(`stderr: ${result.stderr}`);  // Add this line
   }
   ```

2. **Include stderr in the error event** sent to CLI so users can see what failed

3. **Consider**: Always log sandbox stderr at info level since it contains important execution details

## Environment

- Production deployment
- Model: minimax-m2  
- Agent: standard-agent

--- a/turbo/apps/web/app/api/agent/runs/route.ts
+++ b/turbo/apps/web/app/api/agent/runs/route.ts
@@ -240,13 +240,23 @@ export async function POST(request: NextRequest) {
         console.log(`[API] Run ${run.id} completed successfully`);
       })
       .catch(async (error) => {
+        // Extract error message - E2B CommandExitError includes result with stderr
+        let errorMessage =
+          error instanceof Error ? error.message : "Unknown error";
+
+        // Check if error has result property (E2B CommandExitError)
+        const errorWithResult = error as { result?: { stderr?: string } };
+        if (errorWithResult.result?.stderr) {
+          errorMessage = errorWithResult.result.stderr;
+        }
+
         // Update run with error on failure
-        console.error(`[API] Run ${run.id} failed:`, error);
+        console.error(`[API] Run ${run.id} failed:`, errorMessage);
         await globalThis.services.db
           .update(agentRuns)
           .set({
             status: "failed",
-            error: error instanceof Error ? error.message : "Unknown error",
+            error: errorMessage,
             completedAt: new Date(),
           })
           .where(eq(agentRuns.id, run.id));
@@ -254,7 +264,7 @@ export async function POST(request: NextRequest) {
         // Send vm0_error event
         await sendVm0ErrorEvent({
           runId: run.id,
-          error: error instanceof Error ? error.message : "Unknown error",
+          error: errorMessage,
           errorType: "sandbox_error",
         });
       });

--- a/turbo/apps/web/src/lib/e2b/e2b-service.ts
+++ b/turbo/apps/web/src/lib/e2b/e2b-service.ts
@@ -234,10 +234,19 @@ export class E2BService {
     } catch (error) {
       const executionTimeMs = Date.now() - startTime;
       const completedAt = new Date();
-      const errorMessage =
+
+      // Extract error message - E2B CommandExitError includes result with stderr
+      let errorMessage =
         error instanceof Error ? error.message : "Unknown error";
 
-      log.error(`Run ${context.runId} failed:`, error);
+      // Check if error has result property (E2B CommandExitError)
+      const errorWithResult = error as { result?: { stderr?: string } };
+      if (errorWithResult.result?.stderr) {
+        errorMessage = errorWithResult.result.stderr;
+        log.error(`Run ${context.runId} failed with stderr:`, errorMessage);
+      } else {
+        log.error(`Run ${context.runId} failed:`, error);
+      }
 
       // Send vm0_error event so CLI doesn't timeout
       try {
@@ -503,6 +512,7 @@ export class E2BService {
       log.debug(`Run ${runId} completed successfully`);
     } else {
       log.error(`Run ${runId} failed with exit code ${result.exitCode}`);
+      log.error(`stderr: ${result.stderr}`);
     }
 
     return {

